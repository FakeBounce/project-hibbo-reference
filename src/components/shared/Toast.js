import React from 'react';
import { Image, Animated } from 'react-native';
import { string, func, number } from 'prop-types';
import appStyle from 'styles/appStyles';
import toast from 'assets/auth/icons/toast.png';
import {
  SToastGlobalContainer,
  SToastText,
  SToastContainer,
  SToastContent,
  SToastImageContainer,
  SToastTextContainer,
  SToastTriangle,
} from './styles';

const Toast = ({ label, position, color, hideToast, hideDelay }) => {
  if (hideToast !== null) {
    // Func to hide toast within some time
    Animated.delay(hideDelay).start(hideToast);
  }
  return (
    <SToastGlobalContainer>
      <SToastContainer color={color} position={position}>
        <SToastContent>
          <SToastImageContainer>
            <Image source={toast} />
          </SToastImageContainer>
          <SToastTextContainer>
            <SToastText>{label}</SToastText>
          </SToastTextContainer>
        </SToastContent>
      </SToastContainer>
      <SToastTriangle color={color} position={position} />
    </SToastGlobalContainer>
  );
};

Toast.defaultProps = {
  color: appStyle.colors.red,
  hideToast: null,
  hideDelay: 2000,
};

Toast.propTypes = {
  color: string,
  label: string.isRequired,
  position: string.isRequired,
  hideToast: func,
  hideDelay: number,
};

export default Toast;
