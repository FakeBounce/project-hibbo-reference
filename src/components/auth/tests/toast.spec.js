import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import Toast from 'shared/Toast';

describe('Toast', () => {
  it('should render the component with top position', () => {
    const toast = renderer.create(<Toast label="Some label" position="top" />).toJSON();

    expect(toast).toMatchSnapshot();
  });

  it('should render the component with bottom position', () => {
    const toast = renderer.create(<Toast label="Some label" position="bottom" />).toJSON();

    expect(toast).toMatchSnapshot();
  });
});
