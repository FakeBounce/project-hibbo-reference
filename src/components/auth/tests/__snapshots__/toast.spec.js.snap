// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toast should render the component with bottom position 1`] = `
<View
  style={
    Array [
      undefined,
      Object {
        "backgroundColor": "#FF7171",
      },
    ]
  }
>
  <View
    style={undefined}
  >
    <View
      style={undefined}
    >
      <Image
        source={1}
        style={undefined}
      />
    </View>
    <View
      style={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={undefined}
      >
        Some label
      </Text>
    </View>
  </View>
  <View
    style={
      Array [
        undefined,
        undefined,
        Object {
          "borderBottomColor": "#FF7171",
        },
      ]
    }
  />
</View>
`;

exports[`Toast should render the component with top position 1`] = `
<View
  style={
    Array [
      undefined,
      Object {
        "backgroundColor": "#FF7171",
      },
    ]
  }
>
  <View
    style={undefined}
  >
    <View
      style={undefined}
    >
      <Image
        source={1}
        style={undefined}
      />
    </View>
    <View
      style={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={undefined}
      >
        Some label
      </Text>
    </View>
  </View>
  <View
    style={
      Array [
        undefined,
        undefined,
        Object {
          "borderBottomColor": "#FF7171",
        },
      ]
    }
  />
</View>
`;
